statsmodels/stats/tests/test_lilliefors.py

```diff
@@ -0,0 +1,54 @@
+from numpy.testing import assert_, assert_raises, assert_almost_equal
+from scipy import stats
+import nose
+from statsmodels.stats._lilliefors import lilliefors
+import numpy as np
+
+class TestLilliefors(object):
+
+    def test_normal(self):
+        np.random.seed(3975)
+        x_n = stats.norm.rvs(size=500)
+        # R function call:
+        # require(nortest)
+        # lillie.test(x_n)
+        d_ks_norm, p_norm = lilliefors(x_n, dist='norm')
+        # shift normal distribution > 0 to exactly mirror R `KScorrect` test
+        # R `KScorrect` requires all values tested for exponential
+        # distribution to be > 0
+        # R function call:
+        # require(KScorrect)
+        # LcKS(x_n+abs(min(x_n))+0.001, 'pexp')
+        d_ks_exp, p_exp = lilliefors(x_n+np.abs(x_n.min()) + 0.001, dist='exp')
+        # assert normal
+        assert_almost_equal(d_ks_norm, 0.025957, decimal=3)
+        assert_almost_equal(p_norm, 0.2000, decimal=3)
+        # assert exp
+        assert_almost_equal(d_ks_exp, 0.3436007, decimal=3)
+        assert_almost_equal(p_exp, 0.01, decimal=3)
+
+
+    def test_expon(self):
+        np.random.seed(3975)
+        x_e = stats.expon.rvs(size=500)
+        # R function call:
+        # require(nortest)
+        # lillie.test(x_n)
+        d_ks_norm, p_norm = lilliefors(x_e, dist='norm')
+        # R function call:
+        # require(KScorrect)
+        # LcKS(x_e, 'pexp')
+        d_ks_exp, p_exp = lilliefors(x_e, dist='exp')
+        # assert normal
+        assert_almost_equal(d_ks_norm, 0.15581, decimal=3)
+        assert_almost_equal(p_norm, 2.2e-16, decimal=3)
```

Comment:
The normality test uses the a p-value approximation on the low end (p < 0.1), so it's able to return p-values below those specified in the table. 
The exponential test doesn't have a comparable approximation function, so the lower bound is the same as in the table (p = 0.01). 

The cited reference is "Dallal Wilkinson 1986".